### PR TITLE
Add windows CI

### PR DESCRIPTION
**Pull request by liufengyun**
Add windows CI

We need a basic windows CI to prevent regressions (See #9642)

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+matrix:
+  fast_finish: true
+
+init:
+  - cmd: choco install sbt
+  - cmd: refreshenv
+build_script:
+  - cmd: sbt dotty-bootstrapped/compile
+test_script:
+  - cmd: sbt "dotty-bootstrapped/dotc tests\pos\HelloWorld.scala"
```
Comment (sjrd) [inline .appveyor.yml:12]: May I suggest to add
```
  - cmd: sbt sjsJUnitTests/test
```
as an additional test to improve coverage on the Windows CI?

**Review by sjrd — COMMENTED** [review]
Thank you for this, @liufengyun !

File: .appveyor.yml (hunk 1 above)
Comment (liufengyun) [reply .appveyor.yml:12]: It's done now.

File: .appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - cmd: sbt sjsJUnitTests/test
+  - cmd: sbt test
```
Comment (sjrd) [inline .appveyor.yml:14]: `test` won't work on Windows at the moment. Although `run` tests basically work, all neg tests are completely broken on Windows, because they pervasively expect file names with `/`. Fixing those is probably best left to another day, or at least another PR. ;)

File: compiler/src/dotty/tools/io/ZipArchive.scala [hunk 3]
```diff
@@ -199,7 +199,7 @@ final class FileZipArchive(jpath: JPath) extends ZipArchive(jpath) {
 final class ManifestResources(val url: URL) extends ZipArchive(null) {
   def iterator(): Iterator[AbstractFile] = {
     val root     = new DirEntry("/", null)
-    val dirs     = new mutable.HashMap[String, DirEntry]; dirs.put("/", root)
+    val dirs     = mutable.HashMap[String, DirEntry]("/" -> root)
```
Comment (sjrd) [inline compiler/src/dotty/tools/io/ZipArchive.scala:202]: Was this change intended in this PR?

Comment (liufengyun) [reply compiler/src/dotty/tools/io/ZipArchive.scala:202]: This is not intended, I'll move it to #9644.

**Comment by sjrd** [comment]
The JS tests failed because the default Node.js version is too old on the AppVeyor images. You can install a recent enough version with the following lines in `.appveyor.yml`:
https://github.com/scala-js/scala-js/blob/598063ae88b5b26145e968e1990ca677f63d5123/appveyor.yml#L3-L5
and
https://github.com/scala-js/scala-js/blob/598063ae88b5b26145e968e1990ca677f63d5123/appveyor.yml#L8-L9

You may also want to copy some other lines from that file, for example setting options to give more memory to sbt (currently it quickly disables compilation because the code cache is full):
https://github.com/scala-js/scala-js/blob/598063ae88b5b26145e968e1990ca677f63d5123/appveyor.yml#L13

**Review by sjrd — APPROVED** [review]

**Comment by liufengyun** [comment]
Merge, so other PRs can be green.
